src/main/java/uk/gov/pay/adminusers/queue/event/EventMessageHandler.java

```diff
@@ -1,42 +1,122 @@
+                    logger.warn("Unknown event type: {}", message.getEvent().getEventType());
+                }
                 eventSubscriberQueue.markMessageAsProcessed(message.getQueueMessage());
             } catch (Exception e) {
                 Sentry.captureException(e);
-                logger.warn("Error during handling the event message",
+                logger.warn("An error occurred handling the event message",
                         kv("sqs_message_id", message.getQueueMessage().getMessageId()),
                         kv("resource_external_id", message.getEvent().getResourceExternalId()),
                         kv("error", e.getMessage())
                 );
             }
         }
     }
+
+    private void handleDisputeCreatedMessage(Event disputeCreatedEvent) throws JsonProcessingException {
+        var disputeCreatedDetails = objectMapper.treeToValue(disputeCreatedEvent.getEventData(), DisputeCreatedDetails.class);
+
+        Service service = serviceFinder.byExternalId(disputeCreatedEvent.getServiceId())
+                .orElseThrow(() -> new IllegalArgumentException(format("Service not found [id: %s]", disputeCreatedEvent.getServiceId())));
+        LedgerTransaction transaction = ledgerService.getTransaction(disputeCreatedEvent.getParentResourceExternalId())
+                .orElseThrow(() -> new IllegalArgumentException(format("Transaction not found [id: %s]", disputeCreatedEvent.getParentResourceExternalId())));
+        
+        List<UserEntity> serviceUsers = userDao.findByServiceId(service.getId());
+        List<UserEntity> serviceAdmins = serviceUsers.stream().filter(userEntity -> {
+            var hasAdminRole = userEntity.getRoles().stream().filter(RoleEntity::isAdmin).count();
+            return hasAdminRole > 0;
+        }).collect(Collectors.toList());
+
+        var epoch = disputeCreatedDetails.getDisputeEvidenceDueDate();
+        String formattedDueDate = getZDTForEpoch(epoch).format(dateTimeFormatter);
+        String formattedPayDueDate = getPayDueByDateForEpoch(epoch).format(dateTimeFormatter);
+
+        var paymentAmountInPounds = convertPenceToPounds.apply(disputeCreatedDetails.getPaymentAmount()).toString();
+        var disputeFeeInPounds = convertPenceToPounds.apply(disputeCreatedDetails.getDisputeFee()).toString();
+
+        Map<String, String> personalisation = Stream.of(new String[][]{
+                {"serviceName", service.getName()},
+                {"paymentExternalId", disputeCreatedEvent.getParentResourceExternalId()},
+                {"serviceReference", transaction.getReference()},
+                {"paymentAmount", paymentAmountInPounds},
+                {"disputeFee", disputeFeeInPounds},
+                {"disputeEvidenceDueDate", formattedDueDate},
+                {"sendEvidenceToPayDueDate", formattedPayDueDate}
+        }).collect(Collectors.toMap(data -> data[0], data -> data[1]));
+
+        serviceAdmins.removeAll(Collections.singleton(null));
```

Comment:
How would there be nulls in the service admin list?

Reply:
Removed